asset-transfer-private-data/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,412 @@
+    'peerOrganizations',
+    'org2.example.com'
+);
+
+// Path to org2 user private key directory.
+const keyDirectoryPathOrg2 = path.resolve(
+    cryptoPathOrg2,
+    'users',
+    'User1@org2.example.com',
+    'msp',
+    'keystore'
+);
+
+// Path to org2 user certificate.
+const certPathOrg2 = path.resolve(
+    cryptoPathOrg2,
+    'users',
+    'User1@org2.example.com',
+    'msp',
+    'signcerts',
+    'cert.pem'
+);
+
+// Path to org2 peer tls certificate.
+const tlsCertPathOrg2 = path.resolve(
+    cryptoPathOrg2,
+    'peers',
+    'peer0.org2.example.com',
+    'tls',
+    'ca.crt'
+);
+
+// Gateway peer endpoint.
+const peerEndpointOrg1 = 'localhost:7051';
+const peerEndpointOrg2 = 'localhost:9051';
+
+// Gateway peer container name.
+const peerContainerOrg1 = 'peer0.org1.example.com';
+const peerContainerOrg2 = 'peer0.org2.example.com';
+
+const utf8Decoder = new TextDecoder();
+
+//Collection Names
+const org1PrivateCollectionName = 'Org1MSPPrivateCollection';
+const org2PrivateCollectionName = 'Org2MSPPrivateCollection';
+
+const RED = '\x1b[31m\n';
+const RESET = '\x1b[0m';
+
+//Use a random key so that we can run multiple times
+const randomNumber = Math.floor(Math.random() * 1000) + 1;
+const assetID1 = `asset${randomNumber}`;
+const assetID2 = `asset${randomNumber + 1}`;
+
+async function main(): Promise<void> {
+    // The gRPC client connection from org1 should be shared by all Gateway connections to this endpoint.
+    const clientOrg1 = await newGrpcConnection(
+        tlsCertPathOrg1,
+        peerEndpointOrg1,
+        peerContainerOrg1
+    );
+
+    const gatewayOrg1 = connect({
+        client: clientOrg1,
+        identity: await newIdentity(certPathOrg1, mspIdOrg1),
+        signer: await newSigner(keyDirectoryPathOrg1),
+    });
+
+    // The gRPC client connection from org2 should be shared by all Gateway connections to this endpoint.
+    const clientOrg2 = await newGrpcConnection(
+        tlsCertPathOrg2,
+        peerEndpointOrg2,
+        peerContainerOrg2
+    );
+
+    const gatewayOrg2 = connect({
+        client: clientOrg2,
+        identity: await newIdentity(certPathOrg2, mspIdOrg2),
+        signer: await newSigner(keyDirectoryPathOrg2),
+    });
+
+    try {
+        // Get a network instance representing the channel where the smart contract is deployed.
+        const networkOrg1 = gatewayOrg1.getNetwork(channelName);
+
+        // Get the smart contract from the network.
+        const contractOrg1 = networkOrg1.getContract(chaincodeName);
+
+        // Get a network instance representing the channel where the smart contract is deployed.
+        const networkOrg2 = gatewayOrg2.getNetwork(channelName);
+
+        // Get the smart contract from the network.
+        const contractOrg2 = networkOrg2.getContract(chaincodeName);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~');
+
+        // Create new assets on the ledger.
+        await createAsset(contractOrg1);
```

Comment:
This creates two assets. That's fine but to be picky, maybe either:
- the function name should be `createAssets()` instead of `createAsset()`; or
- there should be two separate functions; or
- there should be one `createAsset()` function that additionally takes the asset to create as an argument

Reply:
Actually, I'm not really sure of the value of the second asset. The only thing that happens to it is being created here and then deleted near the end of this function. It doesn't seem to be obviously demonstrating any aspect of private data not already covered. I'd be tempted to remove it to simplify the sample. @denyeart might also have an opinion.